src/GitVersion.Core/VersionCalculation/VersionSearchStrategies/BaseVersion.cs

```diff
@@ -10,13 +10,13 @@ public BaseVersion() : this(new BaseVersionOperand())
     {
     }
 
-    public BaseVersion(string source, SemanticVersion semanticVersion, ICommit? baseVersionSource = null)
-        : this(new BaseVersionOperand(source, semanticVersion, baseVersionSource))
-    {
-    }
+    public BaseVersion(string source, SemanticVersion semanticVersion, ICommit? baseVersionSource = null, VersionIncrementSourceType sourceType = VersionIncrementSourceType.Tree)
+        : this(new BaseVersionOperand(source, semanticVersion, baseVersionSource, sourceType)) => this.SourceType = sourceType;
 
     public string Source => (Operator?.Source).IsNullOrEmpty() ? Operand.Source : Operator.Source;
 
+    public VersionIncrementSourceType SourceType { get; init; } = VersionIncrementSourceType.Tree;
```

Comment:
Can you please explain me for what the enum `VersionIncrementSourceType` is good for? In my opinion the hole SourceType property can be removed.